src/libcore/hash/mod.rs

```diff
@@ -171,6 +171,8 @@ pub trait Hasher {
 #[unstable(feature = "hash_default",
            reason = "not the most ergonomic interface unless `H` is defaulted \
                      to SipHasher, but perhaps not ready to commit to that")]
+#[deprecated(since = "1.3.0",
+             reason = "has yet to prove itself useful")]
 pub fn hash<T: Hash, H: Hasher + Default>(value: &T) -> u64 {
     let mut h: H = Default::default();
     value.hash(&mut h);
```

Comment:
I feel like this function doesn't quite deserve to be here without https://github.com/rust-lang/rfcs/pull/1196, because with that RFC you could write (after swapping the type parameter order)

```rust
hash::<SipHasher>(&foo);
hash::<FnvHasher>(&foo);
```
whereas today you write:

```rust
hash::<_, SipHasher>(&foo);
hash::<_, FnvHasher>(&foo);
```